tasks/main.yml

```diff
@@ -28,6 +28,13 @@
     state: directory
   with_items: "{{ os_images_list }}"
 
+- name: Download the image to the directory when an image_url is defined
+  get_url:
+    url: "{{ item.image_url }}"
+    dest: "{{ os_images_cache }}/{{ item.name }}/{{ item.name }}.qcow2"
```

Comment:
We can support multiple image types if we `s/qcow2/{{ item.type | default('qcow2') }}/`.